stable/k8s-secret-updater/templates/app-deploy.yaml

```diff
@@ -26,7 +26,12 @@ spec:
       annotations:
         checksum/secret: {{ include (print $.Template.BasePath "/app-secret.yaml") . | sha256sum }}
     spec:
+    {{ if .Values.serviceAccount.name }}
       serviceAccountName: {{ .Values.serviceAccount.name }}
+    {{ else if .Values.serviceAccount.create }}
+      serviceAccountName: {{ .Chart.Name }}
+    {{ else }}
```

Comment:
This seems unnecessary? Can just skip to `end`.

Reply:
If `.Values.serviceAccount.name` not set, but we want to create a service account using `.Values.serviceAccount.create: true` it will create service account with `.Chart.name`  value as its name